src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -2150,23 +2150,42 @@ void Hole::addCutType(const CutDimensionSet& dimensions)
     const std::string& name = dimensions.name;
 
     std::vector<std::string>* list;
+    int counter = -1;
+
     switch (thread) {
     case CutDimensionSet::Metric:
         HoleCutTypeMap.emplace(CutDimensionKey("ISOMetricProfile", name), dimensions);
         list = &HoleCutType_ISOmetric_Enums;
+        // search for first entry that contains "deprecated"
+        for (auto it = HoleCutType_ISOmetric_Enums.begin();
+             it != HoleCutType_ISOmetric_Enums.end(); it++) {
+            counter += 1;
+            if ((*it).find("deprecated") != std::string::npos)
+                break;
+        }
         break;
     case CutDimensionSet::MetricFine:
         HoleCutTypeMap.emplace(CutDimensionKey("ISOMetricFineProfile", name), dimensions);
         list = &HoleCutType_ISOmetricfine_Enums;
+        for (auto it = HoleCutType_ISOmetricfine_Enums.begin();

```

Comment:
Writing a helper function
```
int getIndexOfFirstDepricatedEntry( std::vector<std::string>& list) {
....
} 
```
could reduce some code duplication here

Reply:
I also thought so but the helper function + its usage was more code lines than two times the call. And since there will for sure no further thread type, we can assume it will stay with those cases.
But OK, I will try a lambda.